pages/userprofile.tsx

```diff
@@ -0,0 +1,98 @@
+import * as React from 'react'
+import Layout from '../components/Layout'
+import ProfileImageInfo from '../components/ProfileImageInfo'
+import { Lesson } from '../@types/lesson'
+import ProfileLessons from '../components/ProfileLessons'
+import ProfileSubmissions from '../components/ProfileSubmissions'
+import withQueryLoader from '../containers/withQueryLoader'
+import { GET_APP } from '../graphql/queries'
+
+type SessionUser = {
+  username: string
+  name: string
+}
+
+type AppQuery = {
+  queryData: AppInfo
+}
+
+type AppInfo = {
+  session: Session
+  lessons: Lesson[]
+}
+
+type Session = {
+  user: SessionUser
+  submissions: Submission[]
+}
+
+type Submission = {
+  diff: string
+  status: string
+  lessonId: number
+  challengeId: number
+}
+
+const UserProfile: React.FC<AppQuery> = ({ queryData }) => {
+  const userInfo = {
+    username: queryData.session.user.username,
+    firstName: queryData.session.user.name.split(' ')[0],
+    lastName: queryData.session.user.name.split(' ')[1]
+  }
+
+  const lessonInfo = queryData.lessons.map(lesson => {
+    const lessonProgress =
+      queryData.session.submissions.filter(
+        r => r.status === 'passed' && r.lessonId === lesson.id
+      ).length / lesson.challenges.length
```

Comment:
Is there anything that's preventing the student from making another submission when it's already approved? If they can, then this logic breaks.

Reply:
This seems to be logic that is breaking by existing user. From older uses, some challenges may have been deleted causing the challengeId to be `null` breaking this logic.